src/commands/synthetics/run-tests-lib.ts

```diff
@@ -191,53 +176,86 @@ const getTestListBySearchQuery = async (
     suite: `Query: ${testSearchQuery}`,
   }))
 }
-
-export const getTestsList = async (
+export const getTriggerConfigs = async (
   api: APIHelper,
   config: RunTestsCommandConfig,
   reporter: MainReporter,
-  suites: Suite[] = []
-) => {
-  // If "testSearchQuery" is provided, always default to running it.
-  if (config.testSearchQuery) {
-    const testsToTriggerBySearchQuery = await getTestListBySearchQuery(api, config.global, config.testSearchQuery)
-
-    if (testsToTriggerBySearchQuery.length > MAX_TESTS_TO_TRIGGER) {
-      reporter.error(
-        `More than ${MAX_TESTS_TO_TRIGGER} tests returned by search query, only the first ${MAX_TESTS_TO_TRIGGER} will be fetched.\n`
-      )
+  suites?: Suite[]
+): Promise<TriggerConfig[]> => {
+  const defaultTestConfigOverrides = config.global
+  // TODO: Clean up locations as part of SYNTH-12989
+  const testConfigOverridesFromEnv = config.locations?.length ? {locations: config.locations} : {}
+  const testsFromSearchQuery = config.testSearchQuery ? await getTestsFromSearchQuery(api, config, reporter) : []
+  const testsFromTestConfigs = await getTestConfigs(config, reporter, suites)
+
```

Comment:
Let's move this line to another paragraph, with a comment:
```suggestion
  const testsFromTestConfigs = await getTestConfigs(config, reporter, suites)

  // Grab the test defined from the search query. Their config will contain the suite name, and the search query itself.
  const testsFromSearchQuery = config.testSearchQuery ? await getTestsFromSearchQuery(api, config, reporter) : []

```

Also, for later, we might want to split this `getTestsFromSearchQuery` function into two:
- `getTestIdsFromSearchQuery` returning the list of public ids
- `getAdditionalPropertiesFromSearchQuery` returning the suite name, and the search query, or `{}` if no search query is defined.
This would avoid the need for the find later on, and be more straightforward: `const testFromSearchQuery = testsFromSearchQuery.find((test) => test.id === id)`